### PR TITLE
fix(ios): corrupted image on ios 13

### PR DESCRIPTION
**Pull request by l0stpenguin**
### Platforms affected
ios 13.0, 13.1

### Motivation and Context

When taking an image on ios 13 via camera, the path is returned correctly but the temp image is corrupted.
Here is the issue:
https://github.com/apache/cordova-plugin-camera/issues/492

### Description

Upon debugging, i found that the image data in `self.data` in `CDVCamera.m` is being released/corrupted after setting the metadata via [`imagePickerControllerReturnImageResult`](https://github.com/apache/cordova-plugin-camera/blob/master/src/ios/CDVCamera.m#L678)
The solution is to create a copy of the `NSData` and use that to write the image on disk.

### Testing

Open camera on ios 13. Take an image. The image returned should be displayed correctly

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

File: src/ios/CDVCamera.m [hunk 1]
```diff
@@ -670,17 +670,21 @@ - (void)imagePickerControllerReturnImageResult
 {
     CDVPictureOptions* options = self.pickerController.pictureOptions;
     CDVPluginResult* result = nil;
+   
+    NSMutableData *dest_data = [NSMutableData data];
```
Comment (timbru31) [inline src/ios/CDVCamera.m]: please use camelCase for variable names. Besides that, can we find a better name than `dest_data`?

Comment (mevinDhun) [reply src/ios/CDVCamera.m]: @timbru31 i renamed it to `imageDataWithExif`. is that fine for you?

**Comment by mevinDhun** [comment]
> Thanks a lot for the PR and the debugging! Some minor comments from my side.
> 
> Besides that, does the change work on lower iOS versions, too?

Yes i have tested it on ios 12 too.

**Comment by amayvs** [comment]
@mevinDhun @timbru31 Is there any ETA on merging this PR?

**Comment by anubhavberry** [comment]
I've taken the latest from the fork and tried to test on 13.1.3. The photo doesn't show up

**Comment by eastwillie** [comment]
I've tried on 13.1.3 wkWebview, it didn't work.

**Comment by mevinDhun** [comment]
@eastwillie @anubhavberry i have tested it on ios 13.1 and 13.2 and confirm it is working properly.

**Comment by eastwillie** [comment]
@mevinDhun 
Here is my code and after I take a photo, there was an error:
finalize:2667: image destination must have at least one image.
Seems to me that the picture hasn't been saved into tmp somehow? Or I made a config mistake?

`openCamera(camera = true) {
            const options = {
                quality: 50,
                targetWidth: 750,
                targetHeight: 750,
                destinationType: window.Camera.DestinationType.FILE_URI,
                sourceType: camera ? window.Camera.DestinationType.FILE_URI : window.Camera.PictureSourceType.SAVEDPHOTOALBUM,
                encodingType: window.Camera.EncodingType.JPEG,
                mediaType: window.Camera.MediaType.PICTURE,
                allowEdit: true,
                correctOrientation: true,
            };
            navigator.camera.getPicture((imageUri) => {
                this.$emit('upload', {
                    success: true,
                    uri: [imageUri],
                });
            }, () => {
                this.$emit('upload', {
                    success: false,
                    uri: [],
                });
            }, options);
        },`

**Comment by ryaa** [comment]
I have been experience the same problem on iOS 13.2 and the problem seems to be that when we copy the image with metadata (exif data) from the source to destination in imagePickerControllerReturnImageResult method of CDVCamera.m the resulting file is of 0 size and CGImageDestinationFinalize returns false (fails!). If I disable geolocation for camera by setting the below in config.xml and everything works well.
```
<preference name="CameraUsesGeolocation" value="false" />
```

I tried two fixes and both seems to be working:
1) I changed imagePickerControllerReturnImageResult in CDVCamera.m based on this suggestion https://github.com/vlinde/cordova-plugin-camera-with-exif/commit/99465b4c078bac48f26eb28a90737396e7dfca92

I removed
```
NSData* data_content = [self.data mutableCopy];
```
and added instead
```
UIImage *image = [UIImage imageWithData:self.data];
CGImageRef imageRef = image.CGImage;
```
plus I replaced this line
```
 CGImageDestinationAddImageFromSource(destinationImage, sourceImage, 0, (__bridge CFDictionaryRef)self.metadata);
```
with this line
```
CGImageDestinationAddImage(destinationImage , imageRef, (CFDictionaryRef)self.metadata);
```
2) I also tried the fix as suggested by @mevinDhun 

Both fixes seems to be working.

I'm still wondering why CGImageDestinationAddImageFromSource fails while CGImageDestinationAddImage works fine when saving the image with the metadata. Any ideas?

**Comment by eastwillie** [comment]
@ryaa Thanks, I saw a dialog ask permission for location, but less than 1 second and disappeared. That explained it.

**Review by erisu — CHANGES_REQUESTED** [review]
The changes in `package.json` and `plugin.xml` should be reverted. Also the branch will need to be rebased with master as there is conflicting files.

**Comment by erisu** [comment]
Closed by #712

**Pull request closed without merging**